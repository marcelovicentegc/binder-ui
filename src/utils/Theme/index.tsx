import baseStyled, { ThemedStyledInterface } from "styled-components";

export const theme = {
  importFontFamily:
    "@import url('https://fonts.googleapis.com/css?family=IBM+Plex+Sans&display=swap');",
  useFontFamily: "font-family: IBM Plex Sans, sans-serif;",
  color: {
    primaryBlack: "#262626",
    black1: "#202020",
    black2: "#2B2B2B",
    white1: "#F7F7F7",
    white2: "#DCDDE0", // TODO: move to grey
    white3: "#F5F5F5",
    gray1: "#E9E9E9",
    gray2: "#C4C4C4",
    gray3: "#999999",
    gray4: "#4C4C4C",
    gray5: "#D9D9D9",
    gray6: "8C8C8C",
    gray7: "#4D4D4D",
    gray8: "#E6E6E6",
    gray9: "#5C5C5C",
    gray10: "#EFF6EA",
    red1: "#D24654",
    red2: "#BB3B46",
    red3: "#A52F38",
    red4: "#831D23",
    red5: "#831D23",
    orange1: "#EBA248",
    orange2: "#E2873C",
    orange3: "#D86C30",
    orange5: "#C43618",
    yellow1: "#F4DA4E",
    yellow2: "#EFC841",
    yellow3: "#EAB634",
    yellow4: "#DA7F0D",
    green1: "#5DC957",
    green2: "#4DAE48",
    green3: "#3E933A",
    green4: "#276B24",
    green5: "#0F420E",
    blue1: "#6ACEDF",
    blue2: "#59B5CF",
    blue3: "#479DBF",
    blue4: "#3584AF",
    blue5: "#12538F",
    purple1: "#5A4D80",
    rainbowColors: [
      "#CC0000",
      "#F06D00",
      "#E9CD00",
      "#00BE3A",
      "#439AE5",
      "#5400CE",
      "#5E0096",
      "#EF0067"
    ]
  }
};

export type Theme = typeof theme;
export const styled = baseStyled as ThemedStyledInterface<Theme>;
