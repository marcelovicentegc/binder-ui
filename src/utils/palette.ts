export const palette = [
  // Gray scale
  "#E9E9E9",
  "#C4C4C4",
  "#999999",
  "#4C4C4C",
  "#262626",
  // Red scale
  "#D24654",
  "#BB3B46",
  "#A52F38",
  "#831D23",
  // Orange scale
  "#EBA248",
  "#E2873C",
  "#D86C30",
  "#C9441E",
  "#C43618",
  // Yellow scale
  "#F4DA4E",
  "#EFC841",
  "#EAB634",
  "#DA7F0D",
  // Green scale
  "#5DC957",
  "#4DAE48",
  "#3E933A",
  "#276B24",
  "#0F420E",
  // Blue scale
  "#6ACEDF",
  "#59B5CF",
  "#479DBF",
  "#3584AF",
  "#12538F"
];
