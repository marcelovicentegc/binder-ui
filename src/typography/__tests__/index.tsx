import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "../../utils/render";
import {
  Title,
  Title2,
  Subtitle,
  H1,
  H2,
  H3,
  Label1,
  Label2,
  Spotlight,
  Legend,
  Legend2
} from "..";
import { theme } from "../../utils/Theme";

describe("<Title /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Title>Hi, I'm Title</Title>);
    expect(getByText("Hi, I'm Title")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Title>Hi, I'm Title</Title>);
    expect(getByText("Hi, I'm Title")).toMatchSnapshot();
  });
});

describe("<Title2 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Title2>Hi, I'm Title2</Title2>);
    expect(getByText("Hi, I'm Title2")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Title2>Hi, I'm Title2</Title2>);
    expect(getByText("Hi, I'm Title2")).toMatchSnapshot();
  });

  test("matches snapshot when color is passed via props", () => {
    const { getByText } = render(
      <Title2 color={theme.color.red4}>Hi, I'm Title2</Title2>
    );
    expect(getByText("Hi, I'm Title2")).toMatchSnapshot();
  });
});

describe("<Subtitle /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Subtitle>Hi, I'm Subtitle</Subtitle>);
    expect(getByText("Hi, I'm Subtitle")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Subtitle>Hi, I'm Subtitle</Subtitle>);
    expect(getByText("Hi, I'm Subtitle")).toMatchSnapshot();
  });

  test("matches snapshot when color is passed via props", () => {
    const { getByText } = render(
      <Subtitle color={theme.color.red2}>Hi, I'm Subtitle</Subtitle>
    );
    expect(getByText("Hi, I'm Subtitle")).toMatchSnapshot();
  });
});

describe("<H1 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<H1>Hi, I'm H1</H1>);
    expect(getByText("Hi, I'm H1")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<H1>Hi, I'm H1</H1>);
    expect(getByText("Hi, I'm H1")).toMatchSnapshot();
  });
});

describe("<H2 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<H2>Hi, I'm H2</H2>);
    expect(getByText("Hi, I'm H2")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<H2>Hi, I'm H2</H2>);
    expect(getByText("Hi, I'm H2")).toMatchSnapshot();
  });
});

describe("<H3 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<H3>Hi, I'm H3</H3>);
    expect(getByText("Hi, I'm H3")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<H3>Hi, I'm H3</H3>);
    expect(getByText("Hi, I'm H3")).toMatchSnapshot();
  });
});

describe("<Label1 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Label1>Hi, I'm Label1</Label1>);
    expect(getByText("Hi, I'm Label1")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Label1>Hi, I'm Label1</Label1>);
    expect(getByText("Hi, I'm Label1")).toMatchSnapshot();
  });
});

describe("<Label2 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Label2>Hi, I'm Label2</Label2>);
    expect(getByText("Hi, I'm Label2")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Label2>Hi, I'm Label2</Label2>);
    expect(getByText("Hi, I'm Label2")).toMatchSnapshot();
  });
});

describe("<Spotlight /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Spotlight>Hi, I'm Spotlight</Spotlight>);
    expect(getByText("Hi, I'm Spotlight")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Spotlight>Hi, I'm Spotlight</Spotlight>);
    expect(getByText("Hi, I'm Spotlight")).toMatchSnapshot();
  });
});

describe("<Legend /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Legend>Hi, I'm Legend</Legend>);
    expect(getByText("Hi, I'm Legend")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Legend>Hi, I'm Legend</Legend>);
    expect(getByText("Hi, I'm Legend")).toMatchSnapshot();
  });
});

describe("<Legend2 /> test case", () => {
  test("renders text", () => {
    const { getByText } = render(<Legend2>Hi, I'm Legend2</Legend2>);
    expect(getByText("Hi, I'm Legend2")).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { getByText } = render(<Legend2>Hi, I'm Legend2</Legend2>);
    expect(getByText("Hi, I'm Legend2")).toMatchSnapshot();
  });
});
