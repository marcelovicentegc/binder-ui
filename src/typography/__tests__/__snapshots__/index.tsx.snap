// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H1 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: 600;
  font-size: 40px;
}

<span
  class="c0"
>
  Hi, I'm H1
</span>
`;

exports[`<H2 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 32px;
}

<span
  class="c0"
>
  Hi, I'm H2
</span>
`;

exports[`<H3 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 20px;
  line-height: 125%;
}

<span
  class="c0"
>
  Hi, I'm H3
</span>
`;

exports[`<Label1 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: 600;
  font-size: 22px;
}

<span
  class="c0"
>
  Hi, I'm Label1
</span>
`;

exports[`<Label2 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: 500;
  font-size: 16px;
}

<span
  class="c0"
>
  Hi, I'm Label2
</span>
`;

exports[`<Legend /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 14px;
}

<span
  class="c0"
>
  Hi, I'm Legend
</span>
`;

exports[`<Legend2 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 12px;
  line-height: 140.62%;
  color: #999999;
}

<span
  class="c0"
>
  Hi, I'm Legend2
</span>
`;

exports[`<Legend3 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 12px;
  line-height: 140.62%;
  font-weight: 600;
  text-transform: uppercase;
  color: #999999;
}

<span
  class="c0"
>
  Hi, I'm Legend3
</span>
`;

exports[`<Spotlight /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: 600;
  font-size: 18px;
}

<span
  class="c0"
>
  Hi, I'm Spotlight
</span>
`;

exports[`<Subtitle /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 28px;
  color: black;
}

<span
  class="c0"
>
  Hi, I'm Subtitle
</span>
`;

exports[`<Subtitle /> test case matches snapshot when color is passed via props 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-size: 28px;
  color: #BB3B46;
}

<span
  class="c0"
  color="#BB3B46"
>
  Hi, I'm Subtitle
</span>
`;

exports[`<Title /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: bold;
  font-size: 44px;
}

<span
  class="c0"
>
  Hi, I'm Title
</span>
`;

exports[`<Title2 /> test case matches snapshot 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: bold;
  font-size: 36px;
  color: black;
}

<span
  class="c0"
>
  Hi, I'm Title2
</span>
`;

exports[`<Title2 /> test case matches snapshot when color is passed via props 1`] = `
.c0 {
  font-family: IBM Plex Sans,sans-serif;
  font-style: normal;
  font-weight: normal;
  line-height: 140.62%;
  -webkit-letter-spacing: -0.01em;
  -moz-letter-spacing: -0.01em;
  -ms-letter-spacing: -0.01em;
  letter-spacing: -0.01em;
  color: black;
  font-weight: bold;
  font-size: 36px;
  color: #831D23;
}

<span
  class="c0"
  color="#831D23"
>
  Hi, I'm Title2
</span>
`;
